### ISSUE TITLE
Model particles

### ISSUE THREAD

**Issue by Extex101** [minetest/minetest#11122]
Add the option to have particles and particle spawners use a model for the particle

**Comment by LoneWolfHT** [comment]
Doesn't that defeat the purpose of particles?
I'd suggest just using entities

**Comment by appgurueu** [comment]
> Doesn't that defeat the purpose of particles?
> I'd suggest just using entities

Not really. Particles are still way more lightweight than entities (at the expense of some features). If we stuff all entity features in particles, they will however cease to have this advantage.

**Comment by appgurueu** [comment]
Update: Let me add the dreaded SSCSM tag. Rationale is quite simple. I will want to expose scene nodes to SSCSM. If we can make this sufficiently performant, mesh particles are just a way of using these scene nodes.

(If you will, it boils down to: "Let's start doing rendering logic on the client, shall we?")

**Comment by SmallJoker** [comment]
Depends on #14018.
